src/components/ProductCard/ProductDetails.jsx

```diff
@@ -0,0 +1,143 @@
+import "./ProductDetails.css";
+import { Link } from "react-router-dom";
+import { useCart, useAuth, useFilter } from "../../context/";
+
+export const ProductDetails = ({ sproduct }) => {
+  const {
+    imgUrl,
+    title,
+    description,
+    newPrice,
+    oldPrice,
+    discount,
+    itemRating,
+    outOfStock
+  } = sproduct;
+
+  const {
+    state: { myWishlist },
+    productDispatch
+  } = useFilter();
+
+  const getClassName = (outOfStock) => {
+    if (outOfStock) {
+      const className =
```

Comment:
Do follow the abstraction comment here as well. Is der any reason you are assigning it and then returning it? You can directly return right.

Reply:
thanks, updated this